wifistats/index.js

```diff
@@ -0,0 +1,114 @@
+const exec          = require('child_process').exec;
+
+// The tools
+const airport       = require('./airport');
+const procwireless  = require('./procwireless');
+const netsh         = require('./netsh');
+
+let toolInstance;
+let supportWifiStats = true;
+
+/**
+ * Uses all available tools to query for wifi stats, and the one we found to
+ * work we store in global variable {@link toolInstance}, otherwise we set
+ * supportWifiStats to false.
+ * We call all tools in parallel and when all finishes we check the results.
+ *
+ * @return {Promise}
+ */
+function initTools() {
+    return new Promise((resolve, reject) => {
+        Promise.all([
+                getStats(airport)
+                    .then(result => resolve({
+                        tool: airport,
+                        result
+                    }))
+                    .catch(() => {}),
+                getStats(procwireless)
+                    .then(result => resolve({
+                        tool: procwireless,
+                        result
+                    }))
+                    .catch(() => {}),
+                getStats(netsh)
+                    .then(result => resolve({
```

Comment:
I don't think this will work the way you want it to. resolve applies to the outer promise. If your first getsStats call finishes and calls resolve, other invocations of resolve should have no effect. What you probably want to do is just pass back the object without calling resolve so that success handler in Promise.all().then(handler) receives the object.